x/upgrade/test/integration_test.go

```diff
@@ -112,10 +112,9 @@ func (s *UpgradeTestSuite) TestLegacyGovUpgradeFailure() {
 	res, err := testnode.SignAndBroadcastTx(s.ecfg, s.cctx.Context, acc, msg)
 	require.NoError(t, err)
 	require.Equal(t, abci.CodeTypeOK, res.Code) // we're only submitting the tx, so we expect everything to work
-	require.NoError(t, s.cctx.WaitForNextBlock())
 
 	// compare the result after the tx has been executed.
-	finalResult, err := testnode.QueryTx(s.cctx.Context, res.TxHash, false)
+	finalResult, err := s.cctx.WaitForTx(res.TxHash, 100)
```

Comment:
100 seems a bit excessive, as if it actually took 99 blocks, something is very broken, but it would likely not be directly related to this test so definitely blocking on it

Reply:
The 100 is an arbitrary number I used to be sure we waited enough. Since the timeout commit is in the order of ms, I think 100 is nothing, and it would allow us to know that something is really broken if the transaction is not included in any of the 100 blocks.
However, I can use a smaller number if you prefer it. If so, let me know which number you prefer

Reply:
we should expect it be accepted within a few blocks, so something as low as 5 should be sufficient. This change is not blocking